cli/src/utils/type_description.rs

```diff
@@ -0,0 +1,273 @@
+        output.push(')');
+        Ok(output)
+    }
+}
+
+impl TypeDescription for TypeDefBitSequence<PortableForm> {
+    fn type_description(&self, registry: &PortableRegistry) -> color_eyre::Result<String> {
+        let bit_order_type = self.bit_order_type.id.type_description(registry)?;
+        let bit_store_type = self.bit_store_type.id.type_description(registry)?;
+        Ok(format!("BitSequence({bit_order_type}, {bit_store_type})"))
+    }
+}
+
+impl TypeDescription for TypeDefSequence<PortableForm> {
+    fn type_description(&self, registry: &PortableRegistry) -> color_eyre::Result<String> {
+        let type_description = self.type_param.id.type_description(registry)?;
+        Ok(format!("Sequence({type_description})"))
+    }
+}
+
+impl TypeDescription for TypeDefCompact<PortableForm> {
+    fn type_description(&self, registry: &PortableRegistry) -> color_eyre::Result<String> {
+        let type_description = self.type_param.id.type_description(registry)?;
+        Ok(format!("Compact({type_description})"))
+    }
+}
+
+impl TypeDescription for TypeDefArray<PortableForm> {
+    fn type_description(&self, registry: &PortableRegistry) -> color_eyre::Result<String> {
+        let type_description = self.type_param.id.type_description(registry)?;
+        Ok(format!("[{type_description}; {}]", self.len))
+    }
+}
+
+impl TypeDescription for TypeDefPrimitive {
+    fn type_description(&self, _registry: &PortableRegistry) -> color_eyre::Result<String> {
+        Ok(match &self {
+            TypeDefPrimitive::Bool => "bool",
+            TypeDefPrimitive::Char => "char",
+            TypeDefPrimitive::Str => "String",
+            TypeDefPrimitive::U8 => "u8",
+            TypeDefPrimitive::U16 => "u16",
+            TypeDefPrimitive::U32 => "u32",
+            TypeDefPrimitive::U64 => "u64",
+            TypeDefPrimitive::U128 => "u128",
+            TypeDefPrimitive::U256 => "u256",
+            TypeDefPrimitive::I8 => "i8",
+            TypeDefPrimitive::I16 => "i16",
+            TypeDefPrimitive::I32 => "i32",
+            TypeDefPrimitive::I64 => "i64",
+            TypeDefPrimitive::I128 => "i128",
+            TypeDefPrimitive::I256 => "i256",
+        }
+        .into())
+    }
+}
+
+impl TypeDescription for TypeDefVariant<PortableForm> {
+    fn type_description(&self, registry: &PortableRegistry) -> color_eyre::Result<String> {
+        const MIN_VARIANT_COUNT_FOR_TRAILING_COMMA: usize = 100;
+        let add_trailing_comma = self.variants.len() >= MIN_VARIANT_COUNT_FOR_TRAILING_COMMA;
+
+        let mut variants_string = String::new();
+        variants_string.push('{');
+        let mut iter = self.variants.iter().peekable();
+        while let Some(variant) = iter.next() {
+            let variant_string = variant.type_description(registry)?;
+            variants_string.push_str(&variant_string);
+
+            if iter.peek().is_some() || add_trailing_comma {
+                variants_string.push(',');
+            }
```

Comment:
Maybe this is a hint to tweak the parsing of `scale-value`'s to allow trailing commas :)

Reply:
Oh, I was not aware of the handling of trailing commas in scale-value, I just looked at the rust formatter for inspiration.